### PR TITLE
feat: add flipper as a web component

### PR DESCRIPTION
**Pull request by chrisdholt**
# Description
This PR adds Flipper as a web component.

This is a DRAFT PR which is based on #2862 and depends on that PR being accepted and merged.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components/src/flipper/flipper.styles.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+export const FlipperStyles = css`
+    ${display("inline-flex")} :host {
+        width: calc(var(--height-number) * 1px);
+        height: calc(var(--height-number) * 1px);
+        justify-content: center;
+        align-items: center;
+        margin: 0;
+        position: relative;
+        fill: var(--neutral-foreground-rest);
+        color: var--neutral-foreground-rest);
```
Comment (janechu) [inline packages/fast-components/src/flipper/flipper.styles.ts:14]: looks like you forgot this "("

Comment (chrisdholt) [reply packages/fast-components/src/flipper/flipper.styles.ts:14]: Nice catch - I'll update this once the spec is approved 👍

Comment (nicholasrice) [inline packages/fast-components/src/flipper/flipper.styles.ts:14]:
```suggestion
        color: var(--neutral-foreground-rest);
```

File: packages/fast-components/src/flipper/flipper.stories.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { FASTFlipper } from "./";
```
Comment (awentzel) [inline packages/fast-components/src/flipper/flipper.stories.ts:1]: As a naming standard, we already know in these stories, classes, etc, that we're dealing with FAST.  Do we need to replicate FAST in the named imports, interfaces, and class names?  Does this help the consumer?  Could we eliminate the prefixed naming for FAST?  We may be able to reduce file size while increasing maintainability in that it's easy to scan the file and understand what's going on.  Just from my fresh perspective.

File: packages/fast-components/src/flipper/flipper.styles.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    :host(.disabled) {
+        opacity: var(--disabled-opacity);
+        cursor: ${disabledCursor};
+    }
+
+    :host(:hover)::before {
+        background: var(--neutral-fill-stealth-hover);
+        border-color: var(--neutral-outline-hover);
+    }
+
```
Comment (awentzel) [inline packages/fast-components/src/flipper/flipper.styles.ts:53]: nit: extra CRLF

File: packages/fast-components/src/flipper/index.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { customElement } from "@microsoft/fast-element";
```
Comment (awentzel) [inline packages/fast-components/src/flipper/index.ts:1]: when do we choose "customElement" (lowercase 'c') vs "Flipper" (upper case 'F') in our naming standards?  Have we documented our naming expectations?

File: packages/fast-components/src/flipper/flipper.stories.ts (hunk 2 above)
Comment (chrisdholt) [reply packages/fast-components/src/flipper/flipper.stories.ts:1]: This is importing what is exported to the consumer which I think we do want name spaced. If we want to revisit naming of what is being exported I'd like to raise that as an issue in the repo so we can track independently of this PR.

File: packages/fast-components/src/flipper/flipper.template.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { html, when } from "@microsoft/fast-element";
+import { Flipper, FlipperDirection } from "./flipper";
+
+export const FlipperTemplate = html<Flipper>`
+<template
+    role="button"
+    tabindex=${x => (x.hiddenFromAT ? -1 : 0)}
+    aria-hidden=${x => (x.hiddenFromAT ? true : void 0)}
```
[the remark this answers is not in the document]
Comment (chrisdholt) [reply packages/fast-components/src/flipper/flipper.template.ts:8]: this implementation is changing based on feedback from the spec. This won't be here after the fact :).

File: packages/fast-components/src/flipper/index.ts (hunk 4 above)
Comment (chrisdholt) [reply packages/fast-components/src/flipper/index.ts:1]: These are two different things and both rooted existing standards and common practices. `@customElement` is a decorator which defines our element as a custom element - it's standards based. `Flipper` and similarly `FASTFlipper` are classes which is why they are capitalized.

**Comment by chrisdholt** [comment]
> A suggestion, as a fixture, it may not matter, however, shouldn't we conform to accessibility standards? For example, should we be using an H1 followed by H4 elements? If, we write fixtures as we'd expect in production adhering to accessibility standards we'd get feedback sooner.

I think that's fair; I'm following previous convention that exists within our repo. We can also align the naming of our fixture files for cleanliness. If you'd like to create an issue we can pri making these updates based on your recommendations.

File: packages/fast-components/src/flipper/flipper.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            : this.classList.remove("disabled");
+    }
+
+    @attr({ attribute: "aria-hidden", mode: "fromView", converter: booleanConverter })
+    public hiddenFromAT: boolean = true;
+    public hiddenFromATChanged(): void {
+        this.hiddenFromAT === false
+            ? this.setAttribute("tabindex", "0")
+            : this.setAttribute("tabindex", "-1");
+    }
```
Comment (EisenbergEffect) [inline packages/fast-components/src/flipper/flipper.ts:23]: Very cool use of the features here!

File: packages/fast-components/src/flipper/flipper.template.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { html, when } from "@microsoft/fast-element";
+import { Flipper, FlipperDirection } from "./flipper";
+
+export const FlipperTemplate = html<Flipper>`
+<template
+    role="button"
+    aria-disabled=${x => (x.disabled ? true : void 0)}
```
Comment (EisenbergEffect) [inline packages/fast-components/src/flipper/flipper.template.ts:7]: Nice!

**Pull request merged**